pkg/dcgmexporter/parser.go

```diff
@@ -127,3 +150,40 @@ func recordIsCommentOrEmpty(s []string) bool {
 
 	return false
 }
+
+func readConfigMap(kubeClient kubernetes.Interface, c *Config) ([][]string, error) {
+	parts := strings.Split(c.ConfigMapData, ":")
+	if len(parts) != 2 {
+		return nil, fmt.Errorf("Malformed configmap-data: %s", c.ConfigMapData)
+	}
+
+	var cm *corev1.ConfigMap
+	cm, err := kubeClient.CoreV1().ConfigMaps(parts[0]).Get(context.TODO(), parts[1], metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("Could not retrieve ConfigMap '%s': %v", c.ConfigMapData, err)
+	}
+
+	r := csv.NewReader(strings.NewReader(cm.Data["metrics"]))
```

Comment:
I'd suggest to validate that "metrics" are actually present in the map:
if metrics, ok := cm.Data["metrics"]; ok {
...
} else {
Report an error here
}

Reply:
I've added logic and an error if 'metrics' is not present.